### PR TITLE
feat(func): support func erase

### PR DESCRIPTION
**Pull request by hantmac**
- support `erase` func

### CODE REVIEW

**Comment by ngjaying** [comment]
You are so fast, amazing.  Unfortunately, merge is already implemented as name "object_concat". Maybe we can just add erase.

**Comment by hantmac** [comment]
> You are so fast, amazing. Unfortunately, merge is already implemented as name "object_concat". Maybe we can just add erase.

Aha, there is why I have not found `merge` 😢

File: internal/binder/function/funcs_obj.go [hunk 1]
```diff
@@ -155,4 +156,51 @@ func registerObjectFunc() {
...
+			res := make(map[string]interface{})
+			argMap, ok := args[0].(map[string]interface{})
+			if !ok {
+				return fmt.Errorf("the first argument should be map[string]interface{}, got %v", args[0]), false
+			}
+			eraseArray := make([]string, 0)
+			v := reflect.ValueOf(args[1])
+			switch v.Kind() {
+			case reflect.Slice, reflect.Array:
+				array := args[1].([]string)
```
Comment (ngjaying) [inline internal/binder/function/funcs_obj.go:182]: The args[1] is likely to be []any, the type cast here will fail. You can use cast.ToStringSlice here to do the conversion.

**Review by ngjaying — CHANGES_REQUESTED** [review]

File: internal/binder/function/funcs_obj.go (hunk 1 above)
Comment (hantmac) [reply internal/binder/function/funcs_obj.go]: Thanks! Fixed.

Comment (ngjaying) [reply internal/binder/function/funcs_obj.go]: ah, here I mean "github.com/lf-edge/ekuiper/pkg/cast"

Comment (hantmac) [reply internal/binder/function/funcs_obj.go]: > ah, here I mean "github.com/lf-edge/ekuiper/pkg/cast"

Thanks the `good first issue` 😅

**Review by ngjaying — APPROVED** [review]

**Review by Yisaer — APPROVED** [review]

**Pull request merged**